include/swift/Syntax/SyntaxCollection.h

```diff
@@ -173,8 +175,10 @@ class SyntaxCollection : public Syntax {
 
   /// Return a new collection with the element removed at index i.
   SyntaxCollection<CollectionKind, Element> removing(size_t i) const {
-    auto NewLayout = getRaw()->Layout;
-    NewLayout.erase(NewLayout.begin() + i);
+    llvm::SmallVector<RC<RawSyntax>, 16> NewLayout = getRaw()->getLayout();
+    auto iterator = NewLayout.begin();
+    std::advance(iterator, i - 1);
```

Comment:
I'm not sure `i - 1` is correct. What if i == 0?
Also, could you add test cases in `unittests/Syntax/SyntaxCollectionTests.cpp`?